nodewatch/monitoring/__init__.py

```diff
@@ -19,249 +19,249 @@
-        self.last_refresh_time = self.last_reload_time
-
-    def refresh_heights(self):
-        NetworkConnector(self.network_name).update_heights(self.repository.node_descriptors)
-        self.last_refresh_time = datetime.datetime.utcnow()
+	def __init__(self, network_name, title_network_name, version_to_css_class, version_customizations):
+		self.network_name = network_name
+		self.title_network_name = title_network_name
+		self.version_to_css_class = version_to_css_class
+		self.version_customizations = VersionCustomizations(version_customizations)
+
+		self.repository = NetworkRepository(self.network_name)
+		self.last_reload_time = datetime.datetime(2021, 1, 1)
+		self.last_refresh_time = None
+
+	def html_harvesters(self):
+		return render_template(
+			f'{self.network_name}_nodes.html',
+			title=f'{self.title_network_name} Recent Harvesters',
+			descriptors=self.repository.harvester_descriptors,
+			version_to_css_class=self.version_to_css_class)
+
+	def html_nodes(self):
+		return render_template(
+			f'{self.network_name}_nodes.html',
+			title=f'{self.title_network_name} Nodes',
+			descriptors=self.repository.node_descriptors,
+			version_to_css_class=self.version_to_css_class)
+
+	def json_height_chart(self):
+		compatible_node_descriptors = [
+			descriptor for descriptor in self.repository.node_descriptors if 'success' == self.version_to_css_class(descriptor.version)
+		]
+		height_builder = HeightChartBuilder(self.version_customizations, MIN_HEIGHT_CLUSTER_SIZE)
+		height_builder.add_heights(compatible_node_descriptors)
+		height_builder.add_finalized_heights(compatible_node_descriptors)
+		return height_builder.create_chart()
+
+	def json_height_chart_with_metadata(self):
+		return json.dumps({
+			'chart_json': self.json_height_chart(),
+			'last_refresh_time': self.last_refresh_time.strftime(TIMESTAMP_FORMAT)
+		})
+
+	def json_height(self):
+		return json.dumps({'height': self.repository.estimate_height()})
+
+	def reload_all(self, resources_path):
+		nodes_filepath = resources_path / f'{self.network_name}_nodes.json'
+		harvesters_filepath = resources_path / f'{self.network_name}_harvesters.csv'
+		voters_filepath = resources_path / f'{self.network_name}_richlist.csv'
+		all_filepaths = [nodes_filepath, harvesters_filepath, voters_filepath]
+
+		# nodes.json is produced by the network crawl, all other files are derived from it
+		last_crawl_timestamp = nodes_filepath.stat().st_mtime
+		last_crawl_time = datetime.datetime.utcfromtimestamp(last_crawl_timestamp)
+		if self.last_reload_time >= last_crawl_time:
+			log.debug(f'skipping update because crawl ({last_crawl_time}) is not newer than reload ({self.last_reload_time})')
+			return
+
+		if any(filepath.exists() and filepath.stat().st_mtime < last_crawl_timestamp for filepath in all_filepaths):
+			log.debug(f'skipping update because some files have not been updated on disk (last crawl {last_crawl_time}, \
+			'f' last reload {self.last_reload_time})')
```

Comment:
why this reformatting?

Reply:
because the length is too long.

`./monitoring/__init__.py:78:141: E501 line too long (153 > 140 characters)`

Reply:
ignore my message, it was my `autopep8` affected.